pkg/controller/daemon/daemoncontroller.go

```diff
@@ -1095,32 +1094,21 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 		return false, false, false, nil
 	}
 
-	// TODO: Move it to the predicates
-	for _, c := range node.Status.Conditions {
-		if critical {
-			break
-		}
-		// TODO: There are other node status that the DaemonSet should ideally respect too,
-		//       e.g. MemoryPressure, and DiskPressure
-		if c.Type == v1.NodeOutOfDisk && c.Status == v1.ConditionTrue {
-			// the kubelet will evict this pod if it needs to. Let kubelet
-			// decide whether to continue running this pod so leave shouldContinueRunning
-			// set to true
-			shouldSchedule = false
-		}
-	}
-
 	reasons, nodeInfo, err := dsc.simulate(newPod, node, ds)
 	if err != nil {
 		glog.Warningf("DaemonSet Predicates failed on node %s for ds '%s/%s' due to unexpected error: %v", node.Name, ds.ObjectMeta.Namespace, ds.ObjectMeta.Name, err)
 		return false, false, false, err
 	}
 
 	var insufficientResourceErr error
-
 	for _, r := range reasons {
 		glog.V(4).Infof("DaemonSet Predicates failed on node %s for ds '%s/%s' for reason: %v", node.Name, ds.ObjectMeta.Namespace, ds.ObjectMeta.Name, r.GetReason())
 		switch reason := r.(type) {
+		case *util.InvalidNodeConditionError:
```

Comment:
This should be a separate 'subreason' of `PredicateFailureError`, similarly to how other predicates are handled.